docs/NFF-0.1-spec.norg

```diff
@@ -0,0 +1,500 @@
+     -- or when the indentation is manually closed with one of the {# delimiting modifiers}
+
+     @code norg
+     - \
+       The indent segment extension has started.
+
+       I can now include multple paragraphs into this list item.
+
+       @code lua
+       print("Even code blocks are allowed!")
+       @end
+
+       ^ Range-able detached modifiers can be inserted, too.
+
+       ---
+     - another nestable detached modifier ends the indent segment
+     - However, in this example I already closed the indent segment manually with `---`
+     @end
+
+** Tags
+
+*** Carryover Tags
+    @code norg
+    |carryover param1 param2 "multi word param" "another param"
+    @end
+
+*** Ranged Tags
+
+**** Ranged Tags which support nested _Norg_ syntax
+     @code norg
+     #table
+     *heading* | /heading/ | _heading_
+     first | second | third
+     1 | | 3
+     | 2 |
+     #end
+     @end
+
+**** Ranged Tags which do *NOT* support nested _Norg_ syntax
+     @code java
+     @MyAnnotation(name="someName",  value="Hello World")
+     public class TheClass {
+       // ...
+     }
+     @end
+
+** Attached Modifiers
+   We can finally discuss attached modifiers (which originally gave rise to the name of
+   {# detached modifiers} as their natural counter-parts). Here are the general rules for attached
+   modifiers:
+   - an opening modifier may only be preceded by whitespace or punctuation
+   - an opening modifier may _NOT_ be succeeded by whitespace
+   - a closing modifier may _NOT_ be preceded by whitespace
+   - a closing modifier may only be succeeded by whitespace or punctuation
+
+   The following attached modifiers exist and have the respective meaning:
+   - \*bold\*: *bold*
+   - \/italic\/: /italic/
+   - \_underline\_: _underline_
+   - \-strikethrough\-: -strikethrough-
+   - \!spoiler\!: !spoiler!
+   - \^superscript\^: ^superscript^  (when nested into `subscript`, will highlight as an error)
+   - \,subscript\,: ,subscript,  (when nested into `superscript`, will highlight as an error)
+   - \`inline code\`: `inline code`  (disables any nested markup)
+   - \+inline comment\+: +inline comment+  (disables any nested markup)
+   - \$inline math\$: $f(x) = y$  (disables any nested markup)
+   - \=variable\=: =variable=  (disables any nested markup)
+
+*** Link Modifier
+    The link modifier, `:`, is a special modifier type called an _intersecting_ modifier. It puts a
+    twist on the original attached modifier rules because it allows non-whitespace characters in
+    front of the opening or after the closing modifier. This allows the intersection of a
+    non-whitespace string like so:
+    @code norg
+    Ex:*am*:ple
+    @end
+    Without the `:`-link modifier, the above example would not result in a bold segment.
+
+    - | | TODO: currently states in the docs that it must occur in pairs but this is not enforced by
+          the TS parser. I.e. the following is valid: ex:_ample_
```

Comment:
Wdyt about this? Do we want to enforce the link modifier to occur in pairs? I lean on the side of not enforcing that because it would mean typing unnecessary chars and as things stand right now everything in this regard works just fine :+1: